File: csharp/bird-watcher/BirdWatcher.cs
using System;

class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek() => new int[] { 0, 2, 5, 3, 7, 8, 4 };

    public int Today() => birdsPerDay[birdsPerDay.Length - 1];

    public void IncrementTodaysCount()
    {
        birdsPerDay[birdsPerDay.Length - 1]++;
    }

    public bool HasDayWithoutBirds()
    {
        bool noBirdDay = false;
        foreach (var count in birdsPerDay)
        {
            if (count == 0)
            {
                noBirdDay = true;
            }
        }
        return noBirdDay;
    }

    public int CountForFirstDays(int numberOfDays)
    {
        int count = 0;
        for (int i = 0; i < Math.Min(numberOfDays, birdsPerDay.Length); i++)
        {
            count += birdsPerDay[i];
        }
        return count;
    }

    public int BusyDays()
    {
        int count = 0;
        foreach (var perDay in birdsPerDay)
        {
            if (perDay >= 5)
            {
                count++;
            }
        }
        return count;
    }
}

File: julia/bird-watcher/bird-watcher.jl
function today(birds_per_day)
    return birds_per_day[end]
end

function increment_todays_count(birds_per_day)
    birds_per_day[end] += 1
    return birds_per_day
end

function has_day_without_birds(birds_per_day)
    return length(birds_per_day[iszero.(birds_per_day)]) > 0
end

function count_for_first_days(birds_per_day, num_days)
    return sum(birds_per_day[1:num_days])
end

function busy_days(birds_per_day)
    return length([bird for bird in birds_per_day if bird >= 5])
end

function average_per_day(week1, week2)
    return [(a + b) / 2 for (a, b) in zip(week1, week2)]
end
